cinema/views.py

```diff
@@ -19,27 +20,48 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
 
 
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
```

Comment:
The `get_permissions` method in `GenreViewSet` allows 'create' action for authenticated users, which might not be intended. Ensure that only admins can create new genres if that is the requirement.